dev/doc/xml-protocol.md

```diff
@@ -753,3 +754,41 @@ Ex: `status = "Idle"` or `status = "proof: myLemmaName"` or `status = "Dead"`
 </feedback>
 ```
 
+## <a name="highlighting">Highlighting</a>
+
+Proof diffs, which are enabled by the `Diffs` option, highlight differences
+between the current and previous proof states in the displayed output.
+These are represented in output fields in the XML protocol with embedded
+XML-like tags.
+
+There are 4 tags that indicate how the enclosed text should be highlighted:
+- diff.added - added text
+- diff.removed - removed text
+- diff.added.bg - unchanged text in a line that has additions ("bg" for "background")
+- diff.removed.bg - unchanged text in a line that has removals
+
+CoqIDE, Proof General and coqtop currently use 2 shades of green and 2 shades of red
+as the background color for highlights.  Coqtop also applies underlining and strikeout
+highlighting for the sake of the color blind.
+
+For example, `<diff.added>ABC</diff.added>` indicates that "ABC" should be highlighted
+as added text.  Tags can be nested, such as:
+`<diff.added.bg>A <diff.added> + 1</diff.added> + B</diff.added.bg>`.  IDE code
+displaying highlighted strings should maintain a stack for nested tags and the associated
+highlight.  Currently the diff code only nests at most 2 tags deep.
+If an IDE uses other highlights such as text foreground color or italic text, it may
+need to merge the background color with those other highlights to give the desired
+(IDE dependent) behavior.
+
+The current implementations avoid highlighting white space at the beginning or the
+end of a line.  This gives a better appearance.
+
+There may be additional text that is tagged with other tags in the output text.  IDEs probably
+should ignore tags and not display tags they don't recognize.
+
+Tagged output strings are generated from Pp.t's.  Use Pp.tag to highlight a Pp.t using
+one of the tags listed above.  A span of tokens can be marked by using "start.<tag>" on
+the first token and "end.<tag>" on the last token.  (Span markers are needed because a span of
+tokens in the output may not match nesting of layout boxes in the Pp.t.)
+The conversion from the Pp.t to the XML-tagged string replaces the "start.\*" and "end.\*"
+tags with the basic tags.
```

Comment:
This last paragraph is less clear, and I wonder if it is targeted to the same audience. Isn't it some internal doc for people that modify the XML protocol? If yes, maybe it isn't really appropriate here, but if there is no better place, it should at least start with something like *Internal (does not matter to XML protocol users):*

Reply:
Indeed, this is internal doc for people who may want to modify the protocol or just understand the code.  For documentation in general, I think it's helpful to selectively include some text that links across abstractions.  The internals doc is very partial.  I will add some more words as you suggest.